ui/src/components/chat/compose.rs

```diff
@@ -324,19 +326,101 @@ fn get_messages(cx: Scope<ComposeProps>) -> Element {
 
 
 fn get_chatbar(cx: Scope<ComposeProps>) -> Element {
-    logger::trace("rendering chatbar");
+    logger::trace("get_chatbar");
     let state = use_shared_state::<State>(cx)?;
     let data = cx.props.data.clone();
     let loading = data.is_none();
+    let input = use_ref(cx, Vec::<String>::new);
+    let reset_input = use_ref(cx, || false);
+    let active_chat_id = data.as_ref().map(|d| d.active_chat.id);
+
+    let should_clear_input = *reset_input.read();
+
+    if should_clear_input {
+        *reset_input.write_silent() = false;
+        // nasty hack because when using mock data, the charbar would be cleared for every keypress after the first message was sent
+        if STATIC_ARGS.use_mock {
+            state.write();
+        }
+    }
+    
+    let ch = use_coroutine(cx, |mut rx: UnboundedReceiver<(Vec<String>, Option<Uuid>)>| {
+        //to_owned![];
+        async move {
+            let warp_cmd_tx = WARP_CMD_CH.tx.clone();
+            while let Some((msg, chat_id)) = rx.next().await {
+                let conv_id = match chat_id {
+                    Some(c) => c,
+                    None => {
+                        continue;
+                    }
+                };
+
+                // don't send empty messages
+                if msg.is_empty() {
+                    continue;
+                }
+
+                // this is also an empty message
+                if msg.iter().all(|line| line.is_empty())  {
```

Comment:
Should probably trim whitespace before checking if its empty (though in warp it does check there as well but doesnt hurt as a precaution)
```suggestion
                if msg.iter().all(|line| line.trim().is_empty())  {
```